### PR TITLE
For #21648: match Pocket sections on home to designs

### PR DESCRIPTION
**Pull request by eliserichards**
For #21648

Designs: https://www.figma.com/file/am9EJGg6ow29PXWkc9jcOH/MR2?node-id=2891%3A46712

* shadows not cut off by bounds
* header text size
* "powered by" text size 
* "Learn more" removing underline
* remove hard-coded color in recent tabs

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by eliserichards** [comment]
image  image

File: app/src/main/java/org/mozilla/fenix/compose/SectionHeader.kt [hunk 1]
```diff
@@ -41,8 +41,39 @@ fun SectionHeader(
...
+ */
+@Composable
+fun HomeSectionHeader(
+    text: String,
+    modifier: Modifier = Modifier
+) {
+    Text(
+        modifier = modifier,
+        text = text,
+        style = TextStyle(
```
Comment (MozillaNoah) [inline app/src/main/java/org/mozilla/fenix/compose/SectionHeader.kt:58]: We should long term probably make a file of JUST text styles

Comment (eliserichards) [reply app/src/main/java/org/mozilla/fenix/compose/SectionHeader.kt:58]: should I file something for that?

**Review by MozillaNoah — APPROVED** [review]
LGTM. Love seeing more and more of Compose

File: app/src/main/java/org/mozilla/fenix/compose/SectionHeader.kt (hunk 1 above)
Comment (MozillaNoah) [reply app/src/main/java/org/mozilla/fenix/compose/SectionHeader.kt:58]: Sure, I think that makes sense. We might also want to have a Compose Consortium meeting or something haha to discuss how to scale all of this now that it's really being implemented.

Comment (Mugurell) [reply app/src/main/java/org/mozilla/fenix/compose/SectionHeader.kt:58]: @eliserichards @MozillaNoah I've created https://github.com/mozilla-mobile/fenix/issues/21192 and started working on a composables theme but the specs are not finished and abandoned the work for now.

**Pull request merged**